### PR TITLE
Add ability to copy binary  files from staged or ftp 

### PR DESCRIPTION
**Pull request by aerorahul**
This PR enables the functionality to copy files from staged fix directory.
The location of the staged fix directory can be set in the env. by defining the variable `GSI_BINARY_SOURCE_DIR`.

A branch in the GSI [here](https://github.com/aerorahul/GSI/tree/feature/switch-fix-git-submodule) switches the `fix` submodule pointer from gerrit to Github and enables copying of the binary files into the `fix` directory.

A tarball containing GSI binaries is uploaded on the EMC FTP site at: https://ftp.emc.ncep.noaa.gov/static_files/public/gsi as gsi-fix-b90f7b.tgz.

This file will be downloaded if:
1. local source dir is not provided
2. local download is corrupt

This file will not be downloaded on successive runs of cmake in the same build directory if the downloaded file is present and the integrity intact.

It takes about 7-10 minutes to download this file from the FTP server.

### CODE REVIEW

**Review by RussTreadon-NOAA — DISMISSED** [review]
Thank you, @aerorahul , for developing this set of changes.   Let's get them into GSI-fix develop.  Approve

**Comment by RussTreadon-NOAA** [comment]
@aerorahul , compressed tarball `gsi-fix-binar.tgz` in `emcrzdm:/home/ftp/emc/gmb/wd20rt` contains binary GSI fix files from the current head (d118c5c) of `gerrit:GSI-fix/rev2` 

```
[wd20rt@vm-cprk-emcrzdm02 wd20rt]$ pwd
/home/ftp/emc/gmb/wd20rt
[wd20rt@vm-cprk-emcrzdm02 wd20rt]$ dir -l
total 1746972
-rw-r--r--. 1 wd20rt emc 1781875122 Jan 12 19:16 gsi-fix-binary.tgz

```

**Comment by aerorahul** [comment]
> @aerorahul , compressed tarball `gsi-fix-binar.tgz` in `emcrzdm:/home/ftp/emc/gmb/wd20rt` contains binary GSI fix files from the current head (d118c5c) of `gerrit:GSI-fix/rev2`
> 
> ```
> [wd20rt@vm-cprk-emcrzdm02 wd20rt]$ pwd
> /home/ftp/emc/gmb/wd20rt
> [wd20rt@vm-cprk-emcrzdm02 wd20rt]$ dir -l
> total 1746972
> -rw-r--r--. 1 wd20rt emc 1781875122 Jan 12 19:16 gsi-fix-binary.tgz
> ```

I created a binary and hosted it here:
`https://ftp.emc.ncep.noaa.gov/static_files/public/gsi` as `gsi-fix-b90f7b.tgz`
`b90f7b` is the truncated `sha256sum` to check the integrity of the file after it is downloaded.

I am working on the download part.  It is working, but I need to increase the `TIMEOUT` interval as the tar file is quite large (1.7GB)

**Comment by RussTreadon-NOAA** [comment]
As usual, you are a few steps ahead of me.   Thank you, @aerorahul

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  file(ARCHIVE_EXTRACT INPUT ${TAR})
+  message(DEBUG "Download ${TAR} succeeded")
+endif()
+
+# Copy from the downloaded or central storage location
+message(STATUS "Copying GSI binary fix files from ${GSI_BIN_FIX_DIR}")
+file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/Big_Endian)
+foreach(file IN LISTS gsi_binaries)
+  message(DEBUG "copying ... ${file}")
+  file(COPY_FILE ${GSI_BIN_FIX_DIR}/${file} ${CMAKE_CURRENT_SOURCE_DIR}/${file} ONLY_IF_DIFFERENT)
```
Comment (RussTreadon-NOAA) [inline CMakeLists.txt]: Attempts to use `COPY_FILE` on WCOSS2 failed. 
```
-- Copying GSI binary fix files from /lfs/h2/emc/global/noscrub/emc.global/FIX/fix/gsi/20221128
CMake Error at fix/CMakeLists.txt:169 (file):
  file does not recognize sub-command COPY_FILE
```

cmake documentation states that [`COPY_FILE`](https://cmake.org/cmake/help/latest/command/file.html#copy-file) is _New in version 3.21_.   

WCOSS2 only has `cmake/3.18.4` and `cmake/3.20.2`

Modifying `fix/CMakeLists.txt` to use

`file(COPY ${GSI_BIN_FIX_DIR}/${file} DESTINATION ${CMAKE_CURRENT_SOURCE_DIR}/${file})`

worked on WCOSS2.   `file(COPY` may have disadvantages which `file(COPY_FILE` does not have.  I don't know.

Comment (RussTreadon-NOAA) [reply CMakeLists.txt]: Ignore my suggestion.  It does not work.

**Review by RussTreadon-NOAA — APPROVED** [review]
Tested on WCOSS2.   Works!  Approve.

**Pull request merged**